src/main/java/com/minecolonies/coremod/entity/citizen/EntityCitizen.java

```diff
@@ -1452,22 +1457,72 @@ public void onDeath(@NotNull final DamageSource damageSource)
             if (!(citizenJobHandler.getColonyJob() instanceof AbstractJobGuard)
                   && (damageSource != DamageSource.IN_WALL))
             {
-                citizenColonyHandler.getColony().setNeedToMourn(true, citizenData.getName());
+                citizenColonyHandler.getColony().addNeedToMourn(true, citizenData.getName());
             }
+
+            if(citizenColonyHandler.getColony().isCoordInColony(world, getPosition()))
```

Comment:
And only if there is a graveyard in the colony

Reply:
The grave are independent for me. I always have trouble finding where my colonist die. They spawn even if there is no graveyard as a maker where to collect the remains of the colonist. If they are not collected, they will decay and eventually despawn anyway.